cli/lightning-cli.c

```diff
@@ -98,26 +220,38 @@ int main(int argc, char *argv[])
 
 	idstr = tal_fmt(ctx, "lightning-cli-%i", getpid());
 	cmd = tal_fmt(ctx,
-		      "{ \"method\" : \"%s\", \"id\" : \"%s\", \"params\" : [ ",
+		      "{ \"method\" : \"%s\", \"id\" : \"%s\", \"params\" : ",
 		      method, idstr);
 
-	for (i = 2; i < argc; i++) {
-		/* Numbers, bools, objects and arrays are left unquoted,
-		 * and quoted things left alone. */
-		if (strspn(argv[i], "0123456789") == strlen(argv[i])
-		    || streq(argv[i], "true")
-		    || streq(argv[i], "false")
-		    || streq(argv[i], "null")
-		    || argv[i][0] == '{'
-		    || argv[i][0] == '['
-		    || argv[i][0] == '"')
-			tal_append_fmt(&cmd, "%s", argv[i]);
+	if (input == DEFAULT_INPUT) {
+		/* Hacky autodetect; only matters if more than single arg */
+		if (argc > 2 && strchr(argv[2], '='))
```

Comment:
I worry about autodetect and prefer a strict default, but that is only a weak preference.

Reply:
Yeah, me too.  But while keywords are better, ordering is what everyone's used to, so I think we're stuck with it.